src/BusinessLayer/TelemetryReporting.cpp

```diff
@@ -1,215 +1,465 @@
+    {
+        unsigned char cmuPacketPayload[unframedPacketLength];
+        std::memcpy(cmuPacketPayload, packetPayload, unframedPacketLength);
+        cmuPacketPayload[1] = i;
+
+        addChecksum(cmuPacketPayload, CMU_LENGTH);
         unsigned char packet[unframedPacketLength + FRAMING_LENGTH_INCREASE];
-        unsigned int packetLength = frameData(packetPayload, unframedPacketLength, packet);
+        unsigned int packetLength = frameData(cmuPacketPayload, unframedPacketLength, packet);
         communicationService_.sendData(packet, packetLength);
     }
 }
 
-void TelemetryReporting::sendMpptData()
+void TelemetryReporting::sendMppt()
 {
-    for (int mpptDataIndex = 0; mpptDataIndex < NUMBER_OF_MPPTS; ++mpptDataIndex)
-    {
-        const unsigned int unframedPacketLength = MPPT_DATA_LENGTH + CHECKSUM_LENGTH;
-        unsigned char packetPayload[unframedPacketLength];
-        packetPayload[0] = MPPT_DATA_ID;
-        packetPayload[1] = mpptDataIndex;
-        packetPayload[2] = static_cast<unsigned char>(vehicleData_.mpptData[mpptDataIndex].type);
+    qDebug() << "Sending Mppt";
+    const unsigned int unframedPacketLength = KEY_MOTOR_LENGTH + CHECKSUM_LENGTH;
+    unsigned char packetPayload[unframedPacketLength];
 
-        if (vehicleData_.mpptData[mpptDataIndex].type == MpptData::Helianthus)
-        {
-           packetPayload[3] = 0x1F;
-        }
-        else
-        {
-           packetPayload[3] = 0x07;
-        }
-        writeFloatIntoArray(packetPayload, 4, vehicleData_.mpptData[mpptDataIndex].voltageIn);
-        writeFloatIntoArray(packetPayload, 8, vehicleData_.mpptData[mpptDataIndex].currentIn);
-        writeFloatIntoArray(packetPayload, 12, vehicleData_.mpptData[mpptDataIndex].voltageOut);
-        writeFloatIntoArray(packetPayload, 16, vehicleData_.mpptData[mpptDataIndex].currentOut);
-        packetPayload[20] = static_cast<unsigned char>(vehicleData_.mpptData[mpptDataIndex].mode);
+    packetPayload[0] = CcsDefines::KEY_MOTOR_PKG_ID;
+    writeUShortIntoArray(packetPayload, 2, mpptData_.arrayVoltage);
+    writeUShortIntoArray(packetPayload, 4, mpptData_.arrayCurrent);
+    writeUShortIntoArray(packetPayload, 6, mpptData_.batteryVoltage);
+    writeUShortIntoArray(packetPayload, 8, mpptData_.temperature);
 
-        addChecksum(packetPayload, MPPT_DATA_LENGTH);
+    for (unsigned char i = 0; i < CcsDefines::MPPT_COUNT; i++)
+    {
+        unsigned char mpptPacketPayload[unframedPacketLength];
+        std::memcpy(mpptPacketPayload, packetPayload, unframedPacketLength);
+        mpptPacketPayload[1] = i;
+
+        addChecksum(mpptPacketPayload, KEY_MOTOR_LENGTH);
         unsigned char packet[unframedPacketLength + FRAMING_LENGTH_INCREASE];
-        unsigned int packetLength = frameData(packetPayload, unframedPacketLength, packet);
+        unsigned int packetLength = frameData(mpptPacketPayload, unframedPacketLength, packet);
         communicationService_.sendData(packet, packetLength);

```

Comment:
Created new issues for new recruits, make data variable for CMUs, MPPTs, and MotorDetails.